src/execution/sema/sema_builtin.cpp

```diff
@@ -1094,47 +1094,55 @@ void Sema::CheckBuiltinTableIterCall(ast::CallExpr *call, ast::Builtin builtin)
 }
 
 void Sema::CheckBuiltinTableIterParCall(ast::CallExpr *call) {
-  if (!CheckArgCount(call, 4)) {
+  if (!CheckArgCount(call, 3)) {
     return;
   }
 
   const auto &call_args = call->Arguments();
 
   // First argument is table name as a string literal
-  if (!call_args[0]->IsStringLiteral()) {
+  if (!call_args[0]->IsIntegerLiteral()) {
     ReportIncorrectCallArg(call, 0, ast::StringType::Get(GetContext()));
     return;
   }
 
-  // Second argument is an opaque query state. For now, check it's a pointer.
-  const auto void_kind = ast::BuiltinType::Nil;
-  if (!call_args[1]->GetType()->IsPointerType()) {
-    ReportIncorrectCallArg(call, 1, GetBuiltinType(void_kind)->PointerTo());
-    return;
-  }
-
-  // Third argument is the thread state container
-  const auto tls_kind = ast::BuiltinType::ThreadStateContainer;
-  if (!IsPointerToSpecificBuiltin(call_args[2]->GetType(), tls_kind)) {
-    ReportIncorrectCallArg(call, 2, GetBuiltinType(tls_kind)->PointerTo());
-    return;
-  }
+  // TODO(Ron): Update type checks here when the argument types are finalized
+  //  // Second argument is an opaque query state. For now, check it's a pointer.
+  //  const auto void_kind = ast::BuiltinType::Nil;
+  //  if (!call_args[1]->GetType()->IsPointerType()) {
```

Comment:
It looks like this definition of the arguments is not being used, in which case, it would be cleaner to delete it